src/libraries/System.Data.Common/ref/System.Data.Common.cs

```diff
@@ -2405,6 +2405,24 @@ public abstract partial class DbDataRecord : System.ComponentModel.ICustomTypeDe
         System.ComponentModel.PropertyDescriptorCollection System.ComponentModel.ICustomTypeDescriptor.GetProperties(System.Attribute[]? attributes) { throw null; }
         object System.ComponentModel.ICustomTypeDescriptor.GetPropertyOwner(System.ComponentModel.PropertyDescriptor? pd) { throw null; }
     }
+    public abstract class DbDataSource : IDisposable, IAsyncDisposable
+    {
+        public abstract string ConnectionString { get; }
+        protected abstract System.Data.Common.DbConnection CreateDbConnection();
+        protected virtual System.Data.Common.DbConnection OpenDbConnection() { throw null; }
+        protected virtual System.Threading.Tasks.ValueTask<System.Data.Common.DbConnection> OpenDbConnectionAsync(System.Threading.CancellationToken cancellationToken = default)  { throw null; }
+        protected virtual System.Data.Common.DbCommand CreateDbCommand(string? commandText = null) { throw null; }
+        protected virtual System.Data.Common.DbBatch CreateDbBatch() { throw null; }
+        public System.Data.Common.DbConnection CreateConnection() { throw null; }
+        public System.Data.Common.DbConnection OpenConnection() { throw null; }
+        public System.Threading.Tasks.ValueTask<System.Data.Common.DbConnection> OpenConnectionAsync() { throw null; }
+        public System.Data.Common.DbCommand CreateCommand(string? commandText = null) { throw null; }
+        public System.Data.Common.DbBatch CreateBatch() { throw null; }
+        public void Dispose() { throw null; }
+        public System.Threading.Tasks.ValueTask DisposeAsync() { throw null; }
+        protected virtual void Dispose(bool disposing) { throw null; }
+        protected virtual System.Threading.Tasks.ValueTask DisposeAsyncCore() { throw null; }
```

Comment:
`CoreDisposeAsync`?

Reply:
Just following the [standard pattern](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync) on this.